src/ecalc/libraries/libecalc/common/libecalc/dto/components.py

```diff
@@ -467,14 +469,28 @@ def validate_unique_names(cls, values):
                 if isinstance(fuel_consumer, GeneratorSet):
                     for electricity_consumer in fuel_consumer.consumers:
                         names.append(electricity_consumer.name)
+                if fuel_consumer.fuel is not None:
+                    for fuel_type in fuel_consumer.fuel.values():
+                        # Need to verify that it is a different fuel
+                        if fuel_type is not None and fuel_type not in fuel_types:
+                            fuel_types.append(fuel_type)
+                            fuel_names.append(fuel_type.name)
 
         duplicated_names = get_duplicates(names)
+        duplicated_fuel_names = get_duplicates(fuel_names)
+
         if len(duplicated_names) > 0:
             raise ValueError(
                 "Component names must be unique. Components include asset/ecalc-model, installations,"
                 " generator sets, electricity consumers, fuel consumers and direct emitters."
                 f" Duplicated names are: {', '.join(duplicated_names)}"
             )
+
```

Comment:
I do think that this check is needed though, but we need an additional check/validation at yaml level.